packages/pelagos/src/components/SvgIcon.js

```diff
@@ -2,31 +2,28 @@ import PropTypes from 'prop-types';
 
 import './SvgIcon.less';
 
-/** An SVG icon. */
-const SvgIcon = ({
-	icon: {
-		icon: [width, height, , , path],
-	},
-	className,
-	animation,
-	...props
-}) => (
+const renderFontAwesomeIcon = (className, {icon: [width, height, , , path]}, props) => (
 	<svg
 		{...props}
-		className={'SvgIcon' + (animation ? ' SvgIcon--' + animation : '') + (className ? ' ' + className : '')}
-		style={{width: width / height + 'em'}}
+		className={`SvgIcon${className ? ` ${className}` : ''}`}
+		style={{width: `${width / height}em`}}
 		focusable="false"
-		viewBox={'0 0 ' + width + ' ' + height}>
+		viewBox={`0 0 ${width} ${height}`}>
 		<path fill="currentColor" d={path} />
 	</svg>
 );
 
+/** An SVG icon. */
+// eslint-disable-next-line no-unused-vars -- used to remove "animation" from "props"
+const SvgIcon = ({icon: Icon, className, animation, ...props}) =>
+	Icon.icon ? renderFontAwesomeIcon(className, Icon, props) : <Icon {...props} className={className} />;
+
 SvgIcon.propTypes = {
```

Comment:
Would it make sense to deprecate `SvgIcon`? Assuming you do not want to keep support for font-awesome indefinitely, should use of `SvgIcon` be discouraged?

I understand why it is used in components like `<IconButton />`, which for now needs to support font-awesome and carbon icons. But any direct use of this outside of the pelagos components would either be pointless (if they are supplying a carbon icon) or deprecated (if they are passing a font-awesome icon).

Reply:
I wasn't sure but I see your point, it would help developers to catch a mistake like the one I had above. Thank you.